### ISSUE TITLE
V5 creation , error on npm start

### ISSUE THREAD

**Issue by HekTheLastSavior** [feathersjs/feathers#2734]
### Steps to reproduce

I created a new V5 app and it installed normally.

when I ran it with ```npm run dev``` it ran normally
but crashed when ran with ```npm run start``` with the following error

```
> shx rm -rf lib/ && tsc

node_modules/socket.io/dist/namespace.d.ts:14:133 - error TS2344: Type 'ListenEvents' does not satisfy the constraint 'EventsMap'.

14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                                                                                                       ~~~~~~~~~~~~

  node_modules/socket.io/dist/namespace.d.ts:14:42
    14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                ~~~~~~~~~~~~
    This type parameter might need an `extends EventsMap` constraint.

node_modules/socket.io/dist/namespace.d.ts:15:42 - error TS2344: Type 'ListenEvents' does not satisfy the constraint 'EventsMap'.

15     new_namespace: (namespace: Namespace<ListenEvents, EmitEvents, ServerSideEvents, SocketData>) => void;
                                            ~~~~~~~~~~~~

  node_modules/socket.io/dist/namespace.d.ts:14:42
    14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                ~~~~~~~~~~~~
    This type parameter might need an `extends EventsMap` constraint.


Found 2 errors in the same file, starting at: node_modules/socket.io/dist/namespace.d.ts:14

```

### System configuration

**Module versions** (especially the part that's not working): **"^5.0.0-pre.28"**

**NodeJS version**: v18.4.0

**Operating System**: Manjaro

**Comment by marshallswain** [comment]
Please add details of what config options you selected.

**Comment by HekTheLastSavior** [comment]
same issue with node v16.17.0

```
❯ feathers generate app
? Do you want to use JavaScript or TypeScript? TypeScript
? What is the name of your application? backend
? Write a short description 
? Which HTTP framework do you want to use? KoaJS (recommended)
? What APIs do you want to offer? HTTP (REST), Real-time
? Which package manager are you using? npm
? Which database are you connecting to? Other databases can be added at any time MongoDB
? Enter your database connection string mongodb://localhost:27017/backend
? Which authentication methods do you want to use? Other methods and providers can be added at any time. Email + Password

```

**Comment by HekTheLastSavior** [comment]
I had the same issue on my macbook pro M1 Motero 12.22 .
So it's neither related to the OS nor the Node version.
Any other suggestions ?

**Comment by marshallswain** [comment]
I just generated an app with node v16.15.1 and it started without error.  Could you please try with a different node version.

**Comment by HekTheLastSavior** [comment]
I tried to run on v16.15.1  on Mac, same issue.
Maybe try to run it on a clean npm state.

**Comment by HekTheLastSavior** [comment]
Update:
If I generated the app without selecting the socket.io option the generated code run normally.
Although concluding the obvious but the issue is in the socket package.

**Comment by daffl** [comment]
Looks like this is an error in TypeScript 4.8.2 - it works properly with version 4.7.4. It also works setting the `skipLibCheck` option to `true` which might open another can of worms but could also be helpful because library definitions (and apparently even TypeScript itself) tend to break *a lot*.

**Comment by marshallswain** [comment]
Thanks @daffl.  I tried upgrading feathers-pinia to 4.8.2 and ran into the same issue last night.  Pinning it at 4.7.4 fixed it.

**Comment by HekTheLastSavior** [comment]
I'll give it a try

**Comment by HekTheLastSavior** [comment]
confirmed to work on typescript@4.7.4 .
will it be fixed to work on later typescript versions ?

**Comment by daffl** [comment]
It appears this is fixed in TypeScript 4.8.3
